src/oge/helpers.py

```diff
@@ -325,11 +389,49 @@ def assign_fuel_category_to_ESC(
     return df
 
 
-def add_plant_local_timezone(df, year):
-    plant_tz = load_data.load_pudl_table(
-        "plants_entity_eia", columns=["plant_id_eia", "timezone"]
+def add_plant_entity(df: pd.DataFrame) -> pd.DataFrame:
+    """Adds timezone and geographical information to input data frame.
+
+    Args:
+        df (pd.DataFrame): table with a 'plant_id_eia' column.
+
+    Returns:
+        pd.DataFrame: original data frame with additional 'timezone', 'lat', 'lon',
+            'state', 'county', 'city', 'zip_code' and 'street_address' columns
+    """
+    geographical_info = [
+        "latitude",
+        "longitude",
+        "state",
+        "county",
+        "city",
+        "zip_code",
+        "street_address",
+    ]
+    plants_entity = load_data.load_pudl_table(
+        "plants_entity_eia", columns=["plant_id_eia", "timezone"] + geographical_info
+    ).astype({"zip_code": "Int64"})
+    plants_entity_from_eia860 = load_data.load_raw_eia860_plant_geographical_info(
+        latest_validated_year
```

Comment:
I get why you use latest_validated_year here, but do we download this file for the latest validated year no matter what year we're running? Just want to make sure we don't run into a filenotfounderror.

Reply:
This is not an issue when running the pipeline since the EIA-860 files are downloaded for the latest validated year in Step 1.

Now if you believe we could use this function outside of the pipeline I can introduce a logic where we check if the file exists and if not is downloaded before loading it.

Reply:
Thanks, this is fine as is then!